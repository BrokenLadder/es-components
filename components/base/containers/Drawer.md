Use ``Drawer`` and ``DrawerPanel`` components to wrap content in collapsable panels.

### Basic Example:
```
<Drawer>
  <DrawerPanel header="Collapsable Item 1">
    <p>Pretty much any content you want in here.</p>
  </DrawerPanel>
  <DrawerPanel header="Collapsable Item 2">
    <ul>
      <li>List Item</li>
      <li>Another Item</li>
    </ul>
  </DrawerPanel>
  <DrawerPanel header="Collapsable Item 3">
    <div>
      More content. Anim pariatur cliche reprehenderit, enim eiusmod
      high life accusamus terry richardson ad squid.
    </div>
  </DrawerPanel>
</Drawer>
```

### Drawer Options

Option            | Type                   | Description
----------------- | ---------------------- | ---------------------------------------------------------
isAccordion       | boolean                | Only allows one DrawerPanel to be open at a time
defaultActiveKeys | string or string array | Specify which panels are opened by default
closedIconName    | string                 | Override the default plus icon with another OE icon name
openedIconName    | string                 | Override the default minus icon with another OE icon name
className         | string or object       | Add additional CSS classes to the root drawer element

### DrawerPanel Options

Option    | Type             | Description
--------- | ---------------- | ---------------------------------------------------------------
header    | string           | Header text displayed next to the open/close icon
key       | string           | Manually set the key value for the panel (otherwise uses index)
className | string or object | Add additional CSS classes to the drawer item element
noPadding | boolean          | Removes the default padding from the panel body


### Advanced Example:
```
<Drawer isAccordion defaultActiveKeys="1" closedIconName="hand-right" openedIconName="hand-down" className="drawer-big">
  <DrawerPanel header="Collapsable Item 1" key="1">
    <p>Pretty much any content you want in here.</p>
  </DrawerPanel>
  <DrawerPanel header="Collapsable Item 2" key="2" className="alt-color">
    <ul>
      <li>List Item</li>
      <li>Another Item</li>
    </ul>
  </DrawerPanel>
  <DrawerPanel header="Collapsable Item 3" key="3" noPadding>
    <div>
      More content. Anim pariatur cliche reprehenderit, enim eiusmod
      high life accusamus terry richardson ad squid.
    </div>
  </DrawerPanel>
</Drawer>
```
