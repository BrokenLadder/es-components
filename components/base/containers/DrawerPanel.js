import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';
import Animate from 'rc-animate';
import cssAnimation from 'css-animation';
import Icon from '../icons/icon';

function animate(node, show, done) {
  let height;
  const animNode = node;
  return cssAnimation(node, 'drawer--animate', {
    start() {
      if (!show) {
        animNode.style.height = `${animNode.offsetHeight}px`;
      } else {
        height = animNode.offsetHeight;
        animNode.style.height = 0;
      }
    },
    active() {
      animNode.style.height = `${show ? height : 0}px`;
    },
    end() {
      animNode.style.height = '';
      done();
    }
  });
}

function animation() {
  return {
    enter(node, done) {
      return animate(node, true, done);
    },
    leave(node, done) {
      return animate(node, false, done);
    }
  };
}

class PanelContent extends Component {
  shouldComponentUpdate(nextProps) {
    return this.props.isActive || nextProps.isActive;
  }

  render() {
    this._isActivated = this._isActivated || this.props.isActive;
    if (!this._isActivated) {
      return null;
    }

    const { isActive, children } = this.props;
    const contentCls = classNames({
      'drawer-panel-body': true,
      'drawer-panel-body--active': isActive,
      'drawer-panel-body--inactive': !isActive
    });

    return (
      <div className={contentCls} role="tabpanel">
        {isActive ? <div className={'drawer-panel-content'}>{children}</div> : null}
      </div>
    );
  }
}

PanelContent.propTypes = {
  isActive: PropTypes.bool,
  children: PropTypes.any
};

const DrawerPanel = (props) => {
  const { className,
    header,
    children,
    isActive,
    openedIconName,
    closedIconName,
    onItemClick } = props;
  const classes = classNames('drawer-panel', className);

  return (
    <div className={classes}>
      <div
        className="drawer-panel-header"
        onClick={() => onItemClick()}
        role="tab"
        aria-expanded={isActive}
      >
        <Icon className="drawer-panel-icon" name={isActive ? openedIconName : closedIconName} />
        {header}
      </div>
      <Animate showProp="isActive" exclusive component="" animation={animation()}>
        <PanelContent isActive={isActive}>
          {children}
        </PanelContent>
      </Animate>
    </div>
  );
};

DrawerPanel.propTypes = {
  children: PropTypes.any,
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  header: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node
  ]),
  isActive: PropTypes.bool,
  onItemClick: PropTypes.func,
  openedIconName: PropTypes.string,
  closedIconName: PropTypes.string
};

DrawerPanel.defaultProps = {
  isActive: false
};

export default DrawerPanel;
