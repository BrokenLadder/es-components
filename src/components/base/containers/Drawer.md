Use ``Drawer`` and ``Drawer.Panel`` components to wrap content in collapsable panels.

### Basic Example:
```
<Drawer>
  <Drawer.Panel title="Panel 1">
    <p>Pretty much any content you want in here.</p>
  </Drawer.Panel>
  <Drawer.Panel title="Panel 2">
    <ul>
      <li>List Item</li>
      <li>Another Item</li>
    </ul>
  </Drawer.Panel>
  <Drawer.Panel title="Panel 3">
    <div>
      More content. Anim pariatur cliche reprehenderit, enim eiusmod
      high life accusamus terry richardson ad squid.
    </div>
  </Drawer.Panel>
</Drawer>
```

### Advanced Example:

Use the ``noPadding`` property to remove default padding within a panel (useful for lists and tables).
You can customize the key property of each Panel; if not specified the default is a zero-based array.
Both the Drawer and Drawer.Panel components will accept additional classNames.

```
<Drawer isAccordion defaultActiveKeys="1" closedIconName="hand-right" openedIconName="hand-down" className="drawer-big">
  <Drawer.Panel title="Panel 1" key="1">
    <p>Pretty much any content you want in here.</p>
  </Drawer.Panel>
  <Drawer.Panel title="Panel 2" key="2" noPadding>
    <ul>
      <li>List Item</li>
      <li>Another Item</li>
    </ul>
  </Drawer.Panel>
  <Drawer.Panel title="Panel 3" key="3" className="alt-color">
    <div>
      More content. Anim pariatur cliche reprehenderit, enim eiusmod
      high life accusamus terry richardson ad squid.
    </div>
  </Drawer.Panel>
</Drawer>
```
