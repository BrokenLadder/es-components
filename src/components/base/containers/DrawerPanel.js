import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../icons/Icon';
import { colors } from '../../theme';
import styled from 'styled-components';
import Collapse from 'react-smooth-collapse-customizable';

const PanelWrapper = styled.div`
  border-bottom: 1px solid ${colors.grayLight};
`;

const PanelTitle = styled.div`
  cursor: pointer;
  padding: 10px 15px;

  &:hover {
    background-color: ${colors.grayLighter};
    color: ${colors.accent};
  }
`;

const PanelIcon = styled(Icon)`
  margin-right: .4em;
  position: relative;
  top: -1px;
`;

const PanelBody = styled(Collapse)`
  background-color: ${colors.white};
  border-bottom: 4px solid ${colors.grayLight};
  color: ${colors.grayDarkest};
  overflow: hidden;

  > div {
    padding: ${(props) => (props.noPadding ? '0' : '10px')};
  }
`;

const DrawerPanel = (props) => {
  const {
    children,
    className,
    closedIconName,
    title,
    isActive,
    noPadding,
    onItemClick,
    openedIconName
  } = props;

  return (
    <PanelWrapper className={className}>
      <PanelTitle onClick={() => onItemClick()} role="tab" aria-expanded={isActive}>
        <PanelIcon name={isActive ? openedIconName : closedIconName} />
        {title}
      </PanelTitle>
      <PanelBody expanded={isActive} heightTransition=".35s ease" noPadding={noPadding}>
        {children}
      </PanelBody>
    </PanelWrapper>
  );
};

DrawerPanel.propTypes = {
  children: PropTypes.any.isRequired,
  /** Add additional CSS classes to the drawer item element */
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  closedIconName: PropTypes.string,
  /** Title text displayed next to the open/close icon */
  title: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node
  ]).isRequired,
  isActive: PropTypes.bool,
  /** Removes the default padding from the panel body */
  noPadding: PropTypes.bool,
  onItemClick: PropTypes.func,
  openedIconName: PropTypes.string
};

DrawerPanel.defaultProps = {
  isActive: false,
  noPadding: false
};

export default DrawerPanel;
