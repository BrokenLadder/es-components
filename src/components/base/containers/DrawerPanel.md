Use ``Drawer`` and ``DrawerPanel`` components to wrap content in collapsable panels.
