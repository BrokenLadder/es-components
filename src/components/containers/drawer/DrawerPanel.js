import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../../base/icons/Icon';
import { colors } from '../../theme';
import styled from 'styled-components';
import Collapse from '../../util/Collapse';
import genId from '../../util/generateAlphaName';

const PanelWrapper = styled.div`
  border-bottom: 1px solid ${colors.grayLight};
`;

const PanelButton = styled.button`
  background: none;
  border: 0;
  cursor: pointer;
  display: flex;
  font-size: 18px;
  justify-content: space-between;
  padding: 10px 15px;
  width: 100%;

  &:hover {
    background-color: ${colors.grayLighter};
    color: ${colors.accent};
  }
`;

const PanelIcon = styled(Icon)`
  margin-right: .4em;
  position: relative;
  top: -1px;
`;

const PanelBody = styled(Collapse)`
  background-color: ${colors.white};
  border-bottom: 4px solid ${colors.grayLight};
  color: ${colors.grayDarkest};

  > div {
    padding: ${props => (props.noPadding ? '0' : '10px 10px 10px 40px')};
  }
`;

const DrawerPanel = props => {
  const {
    children,
    className,
    closedIconName,
    isActive,
    noPadding,
    onItemClick,
    openedIconName,
    title,
    titleAside
  } = props;

  const headingAriaId = genId();
  const regionAriaId = genId();
  const aside = titleAside !== undefined && <aside>{titleAside}</aside>;

  return (
    <PanelWrapper className={className}>
      <div id={headingAriaId} role="heading">
        <PanelButton
          aria-expanded={isActive}
          aria-controls={regionAriaId}
          onClick={() => onItemClick()}
        >
          <span>
            <PanelIcon name={isActive ? openedIconName : closedIconName} />
            {title}
          </span>
          {aside}
        </PanelButton>
      </div>
      <PanelBody
        aria-labelledby={headingAriaId}
        expanded={isActive}
        heightTransition=".35s ease"
        id={regionAriaId}
        noPadding={noPadding}
        role="region"
      >
        {children}
      </PanelBody>
    </PanelWrapper>
  );
};

DrawerPanel.propTypes = {
  children: PropTypes.any.isRequired,
  /** Add additional CSS classes to the drawer item element */
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  closedIconName: PropTypes.string,
  /** Title text displayed next to the open/close icon */
  title: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node
  ]).isRequired,
  /** Aside text/content displayed on the right side of the panel title */
  titleAside: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  isActive: PropTypes.bool,
  /** Removes the default padding from the panel body */
  noPadding: PropTypes.bool,
  onItemClick: PropTypes.func,
  openedIconName: PropTypes.string
};

DrawerPanel.defaultProps = {
  isActive: false,
  noPadding: false
};

export default DrawerPanel;
