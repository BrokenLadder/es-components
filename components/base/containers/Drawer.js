import React, { PropTypes, Children, Component } from 'react';
import classNames from 'classnames';
import './drawer.less';

function toArray(activeKey) {
  let currentActiveKey = activeKey;
  if (!Array.isArray(currentActiveKey)) {
    currentActiveKey = currentActiveKey ? [currentActiveKey] : [];
  }
  return currentActiveKey;
}

class Drawer extends Component {
  constructor(props) {
    super(props);

    const currentActiveKey = this.props.defaultActiveKeys;
    this.state = {
      activeKey: toArray(currentActiveKey)
    };
  }

  componentWillReceiveProps(nextProps) {
    if ('activeKey' in nextProps) {
      this.setState({
        activeKey: toArray(nextProps.activeKey)
      });
    }
  }

  onClickItem(key) {
    return () => {
      let activeKey = this.state.activeKey;

      if (this.props.isAccordion) {
        activeKey = activeKey[0] === key ? [] : [key];
      } else {
        activeKey = [...activeKey];
        const index = activeKey.indexOf(key);
        const isActive = index > -1;
        if (isActive) {
          activeKey.splice(index, 1);
        } else {
          activeKey.push(key);
        }
      }

      this.setState({ activeKey });
    };
  }

  getItems() {
    const activeKey = this.state.activeKey;
    const { isAccordion } = this.props;
    const newChildren = [];

    Children.forEach(this.props.children, (child, index) => {
      if (!child) return;
      // If there is no key provided, use the panel order as default key
      const key = child.key || String(index);
      const header = child.props.header;

      let isActive = false;
      if (isAccordion) {
        isActive = activeKey[0] === key;
      } else {
        isActive = activeKey.indexOf(key) > -1;
      }

      const props = {
        key,
        header,
        isActive,
        children: child.props.children,
        onItemClick: this.onClickItem(key).bind(this),
        closedIconName: this.props.closedIconName,
        openedIconName: this.props.openedIconName
      };

      newChildren.push(React.cloneElement(child, props));
    });

    return newChildren;
  }

  render() {
    const { className } = this.props;
    const classes = classNames('drawer', className);

    return (
      <div className={classes}>
        {this.getItems()}
      </div>
    );
  }
}

Drawer.propTypes = {
  children: PropTypes.any,
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  closedIconName: PropTypes.string,
  defaultActiveKeys: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string)
  ]),
  isAccordion: PropTypes.bool,
  openedIconName: PropTypes.string
};

Drawer.defaultProps = {
  closedIconName: 'plus',
  isAccordion: false,
  openedIconName: 'minus'
};

export default Drawer;
