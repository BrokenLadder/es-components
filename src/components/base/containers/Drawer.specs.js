/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Drawer from './Drawer';
import DrawerPanel from './DrawerPanel';

jest.mock('../../../styles/variables/colors.less', () => ({}));
jest.mock('./drawer.less', () => ({}));

describe('drawer component', () => {
  describe('drawer', () => {
    let drawerInstance;

    beforeEach(() => {
      drawerInstance = mount(<Drawer className="important">
        <DrawerPanel header="collapse 1" key="1">first</DrawerPanel>
        <DrawerPanel header="collapse 2" noPadding key="2">second</DrawerPanel>
        <DrawerPanel header="collapse 3" key="3" className="alt-drawer">third</DrawerPanel>
      </Drawer>);
    });

    it('renders as expected', () => {
      const tree = renderer.create(
        <Drawer className="important">
          <DrawerPanel header="collapse 1" key="1">first</DrawerPanel>
          <DrawerPanel header="collapse 2" noPadding key="2">second</DrawerPanel>
          <DrawerPanel header="collapse 3" key="3" className="alt-drawer">third</DrawerPanel>
        </Drawer>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('adds additional className to the root element', () => {
      const drawerDiv = drawerInstance.find('.drawer');
      expect(drawerDiv.hasClass('important')).toBe(true);
    });

    it('creates the panels closed by default', () => {
      const panels = drawerInstance
                      .find('[aria-expanded]')
                      .filterWhere(node => !node.prop('aria-expanded'));

      expect(panels.length).toBe(3);
    });

    it('adds className to a panel element', () => {
      const drawerPanel = drawerInstance.find('.drawer-panel').last();
      expect(drawerPanel.hasClass('alt-drawer')).toBe(true);
    });

    it('click should toggle panel state', () => {
      const firstPanel = drawerInstance.find('.drawer-panel').first();
      const panelHeader = firstPanel.find('.drawer-panel__header');

      expect(panelHeader.prop('aria-expanded')).toBe(false);

      panelHeader.simulate('click');
      expect(panelHeader.prop('aria-expanded')).toBe(true);

      panelHeader.simulate('click');
      expect(panelHeader.prop('aria-expanded')).toBe(false);
    });

    it('should allow more than one drawer to be opened at a time', () => {
      const firstPanel = drawerInstance.find('.drawer-panel__header').first();
      const lastPanel = drawerInstance.find('.drawer-panel__header').last();

      expect(firstPanel.prop('aria-expanded')).toBe(false);
      expect(lastPanel.prop('aria-expanded')).toBe(false);

      firstPanel.find('.drawer-panel__header').simulate('click');
      expect(firstPanel.prop('aria-expanded')).toBe(true);
      expect(lastPanel.prop('aria-expanded')).toBe(false);

      lastPanel.find('.drawer-panel__header').simulate('click');
      expect(firstPanel.prop('aria-expanded')).toBe(true);
      expect(lastPanel.prop('aria-expanded')).toBe(true);
    });

    it('should set multiple default panels open at start', () => {
      const drawerKeysInstance = mount(<Drawer defaultActiveKeys={['2', '3']}>
        <DrawerPanel header="collapse 1" key="1">first</DrawerPanel>
        <DrawerPanel header="collapse 2" key="2">second</DrawerPanel>
        <DrawerPanel header="collapse 3" key="3">third</DrawerPanel>
      </Drawer>);

      const panels = drawerKeysInstance.find('[aria-expanded]');
      expect(panels.at(0).prop('aria-expanded')).toBe(false);
      expect(panels.at(1).prop('aria-expanded')).toBe(true);
      expect(panels.at(2).prop('aria-expanded')).toBe(true);
    });

    it('should remove padding class on panel with noPadding attrib', () => {
      const panels = drawerInstance.find('.drawer-panel__body');
      expect(panels.at(0).find('.drawer-panel__body--padded').length).toBe(1);
      expect(panels.at(1).find('.drawer-panel__body--padded').length).toBe(0);
      expect(panels.at(2).find('.drawer-panel__body--padded').length).toBe(1);
    });
  });

  describe('accordion', () => {
    let accordionInstance;

    beforeEach(() => {
      accordionInstance = mount(<Drawer className="important" isAccordion defaultActiveKeys="1">
        <DrawerPanel header="collapse 1" key="1">first</DrawerPanel>
        <DrawerPanel header="collapse 2" key="2">second</DrawerPanel>
        <DrawerPanel header="collapse 3" key="3" className="important">third</DrawerPanel>
      </Drawer>);
    });

    it('renders as expected', () => {
      const tree = renderer.create(
        <Drawer className="important" isAccordion defaultActiveKeys="1">
          <DrawerPanel header="collapse 1" key="1">first</DrawerPanel>
          <DrawerPanel header="collapse 2" key="2">second</DrawerPanel>
          <DrawerPanel header="collapse 3" key="3" className="important">third</DrawerPanel>
        </Drawer>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('default active key should set panel open', () => {
      const panels = accordionInstance.find('[aria-expanded]');

      expect(panels.at(0).prop('aria-expanded')).toBe(true);
      expect(panels.at(1).prop('aria-expanded')).toBe(false);
      expect(panels.at(2).prop('aria-expanded')).toBe(false);
    });

    it('should only allow one drawer to be opened at a time', () => {
      const firstPanel = accordionInstance.find('.drawer-panel__header').first();
      const lastPanel = accordionInstance.find('.drawer-panel__header').last();

      expect(firstPanel.prop('aria-expanded')).toBe(true);
      expect(lastPanel.prop('aria-expanded')).toBe(false);

      lastPanel.find('.drawer-panel__header').simulate('click');
      expect(firstPanel.prop('aria-expanded')).toBe(false);
      expect(lastPanel.prop('aria-expanded')).toBe(true);
    });
  });
});
