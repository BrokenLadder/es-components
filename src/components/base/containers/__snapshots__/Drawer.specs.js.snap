exports[`drawer component accordion renders as expected 1`] = `
<div
  className="drawer important">
  <div
    className="drawer-panel">
    <div
      aria-expanded={true}
      className="drawer-panel__header"
      onClick={[Function]}
      role="tab">
      <i
        aria-hidden={true}
        className="drawer-panel__icon hHXwPy" />
      collapse 1
    </div>
    <div
      className="drawer-panel__body drawer-panel__body--padded"
      style={
        Object {
          "display": null,
          "height": "auto",
          "overflow": "hidden",
          "transition": "height .35s ease",
        }
      }>
      <div
        className={undefined}
        style={
          Object {
            "overflow": "hidden",
          }
        }>
        first
      </div>
    </div>
  </div>
  <div
    className="drawer-panel">
    <div
      aria-expanded={false}
      className="drawer-panel__header"
      onClick={[Function]}
      role="tab">
      <i
        aria-hidden={true}
        className="drawer-panel__icon cmlWEk" />
      collapse 2
    </div>
    <div
      className="drawer-panel__body drawer-panel__body--padded"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "overflow": "hidden",
          "transition": "height .35s ease",
        }
      } />
  </div>
  <div
    className="drawer-panel important">
    <div
      aria-expanded={false}
      className="drawer-panel__header"
      onClick={[Function]}
      role="tab">
      <i
        aria-hidden={true}
        className="drawer-panel__icon cmlWEk" />
      collapse 3
    </div>
    <div
      className="drawer-panel__body drawer-panel__body--padded"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "overflow": "hidden",
          "transition": "height .35s ease",
        }
      } />
  </div>
</div>
`;

exports[`drawer component drawer renders as expected 1`] = `
<div
  className="drawer important">
  <div
    className="drawer-panel">
    <div
      aria-expanded={false}
      className="drawer-panel__header"
      onClick={[Function]}
      role="tab">
      <i
        aria-hidden={true}
        className="drawer-panel__icon cmlWEk" />
      collapse 1
    </div>
    <div
      className="drawer-panel__body drawer-panel__body--padded"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "overflow": "hidden",
          "transition": "height .35s ease",
        }
      } />
  </div>
  <div
    className="drawer-panel">
    <div
      aria-expanded={false}
      className="drawer-panel__header"
      onClick={[Function]}
      role="tab">
      <i
        aria-hidden={true}
        className="drawer-panel__icon cmlWEk" />
      collapse 2
    </div>
    <div
      className="drawer-panel__body"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "overflow": "hidden",
          "transition": "height .35s ease",
        }
      } />
  </div>
  <div
    className="drawer-panel alt-drawer">
    <div
      aria-expanded={false}
      className="drawer-panel__header"
      onClick={[Function]}
      role="tab">
      <i
        aria-hidden={true}
        className="drawer-panel__icon cmlWEk" />
      collapse 3
    </div>
    <div
      className="drawer-panel__body drawer-panel__body--padded"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "overflow": "hidden",
          "transition": "height .35s ease",
        }
      } />
  </div>
</div>
`;
